sotodlib/toast/sim_hwpss.py

```diff
@@ -97,30 +109,35 @@ def exec(self, data):
                 )
                 
                 # Scale HWPSS for observing elevation
-
                 el_ref = np.radians(50)
                 scale = np.sin(el_ref) / np.sin(el)
+
+                # Interpolate elevation scaling factor for fast sampling
+                fast_scale = np.interp(fast_time,times,scale)
                 
                 # Observe HWPSS with the detector
 
                 iquv = (transmission + reflection).T
-                iquss = (
-                    iweights * np.interp(chi, self.chis, iquv[0]) +
-                    qweights * np.interp(chi, self.chis, iquv[1]) +
-                    uweights * np.interp(chi, self.chis, iquv[2])
-                ) * scale
+                fast_iquss = (
+                    iweights * np.interp(fast_chi, self.chis, iquv[0]) +
+                    qweights * np.interp(fast_chi, self.chis, iquv[1]) +
+                    uweights * np.interp(fast_chi, self.chis, iquv[2])
+                ) * fast_scale
 
                 iquv = emission.T
-                iquss += (
-                    iweights * np.interp(chi, self.chis, iquv[0]) +
-                    qweights * np.interp(chi, self.chis, iquv[1]) +
-                    uweights * np.interp(chi, self.chis, iquv[2])
+                fast_iquss += (
+                    iweights * np.interp(fast_chi, self.chis, iquv[0]) +
+                    qweights * np.interp(fast_chi, self.chis, iquv[1]) +
+                    uweights * np.interp(fast_chi, self.chis, iquv[2])
                 )
 
-                iquss -= np.median(iquss)
+                fast_iquss -= np.median(fast_iquss)
 
-                # Co-add with the cached signal
+                # Downsample iquss to original sampling rate
+                iquss = scipy.signal.decimate(fast_iquss,10)
```

Comment:
Have you checked that the supersampling and decimation do not introduce a phase shift in the signal? You can do this by applying the same operations to the time stamps and confirming that they match the input time stamps.

Reply:
Decimation no longer required.